peripherals/opamp/opamp_g4.yaml

```diff
@@ -0,0 +1,161 @@
+      Gain64_InputVINM0FilteringVINM1: [29, "Gain 64, input/bias connected to VINM0 with filtering on VINM1 or inverting gain"]   
+    CALSEL:
+      Reference0_033VDDA: [0, "0.033*VDDA applied to OPAMP inputs during calibration"]
+      Reference0_1VDDA: [1, "0.1*VDDA applied to OPAMP inputs during calibration"]
+      Reference0_5VDDA: [2, "0.5*VDDA applied to OPAMP inputs during calibration"]
+      Reference0_9VDDA: [3, "0.9*VDDA applied to OPAMP inputs during calibration"]
+    CALON:
+      NormalMode: [0, "Normal mode"]
+      CalibrationMode: [1, "Calibration mode"]
+    OPAINTOEN:
+      OutputPin: [0, "Output is connected to the output Pin"]
+      ADCChannel: [1, "Output is connected internally to ADC channel"]
+    OPAHSM:
+      NormalMode: [0, "OpAmp in normal mode"]
+      HighSpeedMode: [1, "OpAmp in high speed mode"]
+    VM_SEL:
+      VINM0: [0, "VINM0 connected to VINM input"]
+      VINM1: [1, "VINM1 connected to VINM input"]
+      PGA: [2, "Feedback resistor connected to VINM (PGA mode)"]
+      Output: [3, "OpAmp output connected to VINM (Follower mode)"]
+    USERTRIM:
+      FactoryTrim: [0, "Factory trim used"]
+      UserTrim: [1, "User trim used"]
+    FORCE_VP:
+      NormalMode: [0, "Non-inverting input connected configured inputs"]
+      CalibrationVerificationMode: [1, "Non-inverting input connected to calibration reference voltage"]
+    OPAEN:
+      Enabled: [0, "OpAmp enabled"]
+      Disabled: [1, "OpAmp disabled"]
+
+  OPAMP?_TCMR:
+    LOCK:
+      ReadWrite: [0, "TCMR is read-write"]
+      ReadOnly: [1, "TCMR is read-only, can only be cleared by system reset"]
+    T20CM_EN:
+      AutoSwitchDisabled: [0, "Automatic input switch triggered by TIM20 disabled"]
+      AutoSwitchEnabled: [1, "Automatic input switch triggered by TIM20 enabled"]
+    T8CM_EN:
+      AutoSwitchDisabled: [0, "Automatic input switch triggered by TIM8 disabled"]
+      AutoSwitchEnabled: [1, "Automatic input switch triggered by TIM8 enabled"]
+    T1CM_EN:
+      AutoSwitchDisabled: [0, "Automatic input switch triggered by TIM1 disabled"]
+      AutoSwitchEnabled: [1, "Automatic input switch triggered by TIM1 enabled"]
+
+# OPAMP  VINM0 VINM1 VINP0 VINP1 VINP2 VINP3 VOUT  InternalP InternalOut
+# OPAMP1 PA3   PC5   PA1   PA3   PA7   NONE  PA2   DAC3_CH1  ADC1_IN3, ADC1_IN13
+# OPAMP2 PA5   PC5   PA7   PB14  PB0   PD13  PA6   NONE      ADC2_IN3, ADC2_IN16
+# OPAMP3 PB2   PB10  PB0   PB13  PA1   NONE  PB1   DAC3_CH2  ADC3_IN1, ADC1_IN12, ADC2_IN18, ADC2_IN13
+# OPAMP4 PB10  PD8   PB13  PD11  PB11  NONE  PB12  DAC4_CH1  ADC4_IN3, ADC1_IN11, ADC5_IN5
+# OPAMP5 PB15  PA3   PB14  PD12  PC3   NONE  PA8   DAC4_CH2  ADC5_IN1, ADC_IN3
+# OPAMP6 PA1   PB1   PB12  PD9   PB13  NONE  PB11  DAC3_CH1  ADC12_IN14, ADC4_IN17, ADC3_IN17
+  OPAMP1_CSR:
+    _replace_enum:
+      VM_SEL:
+        PA3_VINM0: [0, "PA3 connected to VINM input"]
```

Comment:
added _VINMx suffix to help people configuring PGA gain figure out what VINMx, etc mean in this context.

Plan to work later on the hal crate implementation, but it will be a week or two until I get around to it. 
I might revisit some of these then.  For now this could be a good start

Reply:
Removed the VINM0 suffix to be in line with stm32f3 opamp definitions